compiler/dyno/include/chpl/framework/Context.h

```diff
@@ -247,6 +255,22 @@ class Context {
   Context();
   ~Context();
 
+  /**
+    Store a string corresponding to CHPL_HOME in the context. The context does
+    not take ownership of the string, so the argument should live at least as
+    long as the context itself.
+
+    Changing chplHome invalidates previous caches of printchplenv output.
+  */
+  void setChplHome(const char* chplHome);
+
+  /**
+    Run printchplenv, or return a cached result of doing so. To get output,
+    CHPL_HOME must have been provided via setChplHome; otherwise, the resulting
+    map will be empty.
+  */
+  llvm::ErrorOr<const ChplEnvMap&> getChplEnv();
```

Comment:
Can you add a comment about how this will handle arguments for adjusting various settings, like `CHPL_COMM=gasnet` for `--comm=gasnet`?

At first I thought they would be arguments here, but instead I think they need to behave like `setChplHome` -- meaning they would be set on the Context.

Reply:
The way that I think the production compiler currently operates is that it fetches this map and then updates keys relevant to itself. Do we want to modify keys in-place on Context's `ChplEnvMap`, or do we want to expect clients to construct a copy (yes, it could be big...) of the map and modify portions of it?

Reply:
> The way that I think the production compiler currently operates is that it fetches this map and then updates keys relevant to itself

I think that's more of the exception than the rule, but I could be wrong. I don't think we should allow in-place mutation on Context's ChplEnvMap. If having the production compiler copy ChplEnvMap helps you implement it, that works for me. The main thing is we don't want to run `printchplenv` twice.

Reply:
Just one last thing: production compiler specifies overrides using a `map<string, const char*>`, but we return the mapping as `unordered_map<string, string>`. Should these be the same? I feel like for Dyno as a library, it makes sense to return strings (and thus explicitly pass ownership without leaking), but it also makes sense for the compiler to pass in a map with `const char*` keys, I guess. 

One wild idea is to make `getChplEnv` in `chplenv.h` be a template function over the input map type...

Reply:
Here's what that might look like: https://github.com/DanilaFe/chapel/commit/2dbdc39f23198f7ab3ef6c0675d197eb01b77849

Reply:
I like the 2-overload approach and a private template implementation in chplenv.cpp. I added this along with a new constructor argument for overriding initial chplevn variables in Context.